tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -354,130 +354,125 @@ def particle_filter(observations,
-    # would force us to trace the state history (with or without thinning),
-    # which is not always appropriate.
-    def seeded_one_step(seed_state_results, _):
-      seed, state, results = seed_state_results
-      one_step_seed, next_seed = samplers.split_seed(seed)
-      next_state, next_results = kernel.one_step(
-          state, results, seed=one_step_seed)
-      return next_seed, next_state, next_results
-
-    final_seed_state_result, traced_results = loop_util.trace_scan(
-        loop_fn=seeded_one_step,
-        initial_state=(loop_seed,
-                       initial_weighted_particles,
-                       kernel.bootstrap_results(initial_weighted_particles)),
-        elems=tf.ones([num_timesteps]),
-        trace_fn=lambda seed_state_results: trace_fn(*seed_state_results[1:]),
-        trace_criterion_fn=(
-            lambda seed_state_results: trace_criterion_fn(  # pylint: disable=g-long-lambda
-                *seed_state_results[1:])),
-        static_trace_allocation_size=static_trace_allocation_size,
-        parallel_iterations=parallel_iterations)
-
-    if trace_criterion_fn is never_trace:
-      # Return results from just the final step.
-      traced_results = trace_fn(*final_seed_state_result[1:])
+        # rejuvenation_fn=...,
+        # rejuvenation_criterion_fn=...,
+        trace_fn=trace_fn,
+    )
 
     return traced_results
 
 
-def _particle_filter_initial_weighted_particles(observations,
-                                                observation_fn,
-                                                initial_state_prior,
-                                                initial_state_proposal,
-                                                num_particles,
-                                                seed=None):
-  """Initialize a set of weighted particles including the first observation."""
-  # Propose an initial state.
-  if initial_state_proposal is None:
-    initial_state = initial_state_prior.sample(num_particles, seed=seed)
-    initial_log_weights = ps.zeros_like(
-        initial_state_prior.log_prob(initial_state))
-  else:
-    initial_state = initial_state_proposal.sample(num_particles, seed=seed)
-    initial_log_weights = (initial_state_prior.log_prob(initial_state) -
-                           initial_state_proposal.log_prob(initial_state))
-  # Normalize the initial weights. If we used a proposal, the weights are
-  # normalized in expectation, but actually normalizing them reduces variance.
-  initial_log_weights = tf.nn.log_softmax(initial_log_weights, axis=0)
-
-  # Return particles weighted by the initial observation.
-  return smc_kernel.WeightedParticles(
-      particles=initial_state,
-      log_weights=initial_log_weights + _compute_observation_log_weights(
-          step=0,
-          particles=initial_state,
-          observations=observations,
-          observation_fn=observation_fn))
-
-
-def _particle_filter_propose_and_update_log_weights_fn(
```

Comment:
Is it possible to keep using `_particle_filter_propose_and_update_log_weights_fn` as-is inside `particle_filter`?